managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -4039,7 +4038,7 @@ public static ManagedLedgerException createManagedLedgerException(Throwable t) {
      */
     protected void asyncCreateLedger(BookKeeper bookKeeper, ManagedLedgerConfig config, DigestType digestType,
             CreateCallback cb, Map<String, byte[]> metadata) {
-        AtomicBoolean ledgerCreated = new AtomicBoolean(false);
+        CompletableFuture<LedgerHandle> ledgerFutureHook = new CompletableFuture<>();
```

Comment:
Why use `CompletableFuture<LedgerHandle>` not `CompletableFuture<Void>`

Reply:
> Why use CompletableFuture<LedgerHandle> not CompletableFuture<Void>

I think `CompletableFuture<LedgerHandle>` is better because it can let us know what it works for and there is no additional cost.